src/content/language/collections.md

```diff
@@ -242,78 +250,180 @@ final constantMap = const {2: 'helium', 10: 'neon', 18: 'argon'};
+
+## Supplementals
 
-### Spread operators
+The following supplemental materials can be used with
+Dart's collection types.
 
-Dart supports the **spread operator** (`...`) and the
-**null-aware spread operator** (`...?`) in list, map, and set literals.
-Spread operators provide a concise way to insert multiple values into a collection.
+<a id="control-flow-operators"></a>
+<a id="spread-operators"></a>
 
-For example, you can use the spread operator (`...`) to insert
-all the values of a list into another list:
+### Spread operations
+
+With a spread operation, you can
+add values from one collection to another collection. Null
+values inside of a  collection can be passed in, but null
+collections produce an error. A spread operation has this
+signature in a collection:
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-spread)"?>
 ```dart
-var list = [1, 2, 3];
-var list2 = [0, ...list];
-assert(list2.length == 4);
+...<collection_name>
 ```
 
-If the expression to the right of the spread operator might be null,
-you can avoid exceptions by using a null-aware spread operator (`...?`):
+In the following example, the values in `a` are added
+as values in `items`.
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-null-spread)"?>
+<?code-excerpt "misc/lib/language_tour/misc/spread_operator_in_collection_a.dart (code_sample)"?>
 ```dart
-var list2 = [0, ...?list];
-assert(list2.length == 1);
+var a = [1, 2, null, 4];
+var items = [0, ...a, 5]; // [0, 1, 2, null, 4, 5]
 ```
 
-For more details and examples of using the spread operator, see the
-[spread operator proposal.][spread proposal]
+In the following example, an error is produced because
+`a` is null and can't be passed into `items`.
+
+```dart
+var a = null;
+var items = [0, ...a, 4]; // Error
+```
+
+To learn more about the operator used in spread operations
+for collections, see [Spread operator][].
+
+[Spread operator]: /language/operators/#spread-operators
+
+### Null-aware spread operations
+
+With a null-aware spread operation, you can add values from
+one collection to another collection. Null values inside of
+a collection can be passed in, but null collections are
+ignored. A null-aware spread operation has this
+signature in a collection:
+
+```dart
+...?<collection_name>
+```
+
+In the following example, `a` is ignored because it's
+null, but `b` is passed into `items`.
+
+```dart
+var a = null;
+var b = [1, 2, null];
+var items = [0, ...?a, ...?b, 4]; // [0, 1, 2, null, 4]
+```
+
+To learn more about the operator used in null-aware spread
+operations for collections, see [Spread operator][].
+
+[Spread operator]: /language/operators/#spread-operators
 
 <a id="collection-operators"></a>
-### Control-flow operators
 
-Dart offers **collection if** and **collection for** for use in list, map,
-and set literals. You can use these operators to build collections using
-conditionals (`if`) and repetition (`for`).
+### If statements
```

Comment:
It depends on how precise/pedantic you want to be, but these technically aren't if statements, they're if elements. The distinction matters somewhat in that the thing inside an if statement's then and else branch is always a statement, and can be any statement:

```dart
if (condition) return x; // Return is a statement.
```

But the thing inside an if element must always be an element:

```dart
var list = [
  if (condition) return x; // Error! Can't have a return here.
];
```

Reply:
This is fascinating. I think we should definitely call them if elements. I'm guessing we call these if key-value pairs for maps? I'll change descriptions and titles so that we can test and see what we think.

Reply:
And after a little research, it seems that a map has elements and an element in a map has a key-value pair in it. I think I'm in alignment now.